### PR TITLE
8344039: Remove security manager dependency in java.time

### PR DESCRIPTION
**Pull request by RogerRiggs**
After [JDK-8338411](https://bugs.openjdk.org/browse/JDK-8338411), java.time implementation dependencies on doPriviledged and AccessController are removed.
Some refactoring to cleanup the remaining code is expected.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344039](https://bugs.openjdk.org/browse/JDK-8344039): Remove security manager dependency in java.time (**Bug** - P4)

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**) Review applies to [0f2e92f5](https://git.openjdk.org/jdk/pull/22042/files/0f2e92f51030cec4d29cd1e194a12f60a21e3993)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**) Review applies to [0f2e92f5](https://git.openjdk.org/jdk/pull/22042/files/0f2e92f51030cec4d29cd1e194a12f60a21e3993)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22042/head:pull/22042` \
`$ git checkout pull/22042`

Update a local copy of the PR: \
`$ git checkout pull/22042` \
`$ git pull https://git.openjdk.org/jdk.git pull/22042/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22042`

View PR using the GUI difftool: \
`$ git pr show -t 22042`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22042.diff">https://git.openjdk.org/jdk/pull/22042.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22042#issuecomment-2471386591)
</details>

### CODE REVIEW

File: src/java.base/share/classes/java/time/zone/ZoneRulesProvider.java [hunk 1]
```diff
@@ -147,27 +145,21 @@ public abstract class ZoneRulesProvider {
         // if the property java.time.zone.DefaultZoneRulesProvider is
         // set then its value is the class name of the default provider
         @SuppressWarnings("removal")
```
Comment (naotoj) [inline src/java.base/share/classes/java/time/zone/ZoneRulesProvider.java:147]: Should this SuppressWarning be removed?

File: src/java.base/share/classes/java/time/zone/ZoneRulesProvider.java
Comment (seanjmullan) [inline src/java.base/share/classes/java/time/zone/ZoneRulesProvider.java:1]: Lines 171-172:

                if (ex.getCause() instanceof SecurityException) {
                    continue;  // ignore the security exception, try the next provider
                }

You may be able to remove this code, but may be better to leave it as-is for now until the SL code has not been checked for cleanup.

**Review by seanjmullan — APPROVED** [review]

**Comment by seanjmullan** [comment]
The PR description above seems to be copied from the serialization PR.

**Review by LanceAndersen — APPROVED** [review]
Looks good to me Roger

**Review by naotoj — APPROVED** [review]
LGTM

**Pull request closed without merging**